src/enochecker/enochecker.py

```diff
@@ -762,9 +771,15 @@ def connect(
         self.debug(
             "Opening socket to {}:{} (timeout {} secs).".format(host, port, timeout)
         )
-        return SimpleSocket(
-            host, port, timeout=timeout, logger=self.logger, timeout_fun=timeout_fun
-        )
+        try:
+            return SimpleSocket(
+                host, port, timeout=timeout, logger=self.logger, timeout_fun=timeout_fun
+            )
+        except socket.error as ex:
+            self.debug(f"Connection to {host}:{port} failed, retrying... {ex}")
+
+        # Not putting this in the exception block - else the stack trace gets looong.
+        return self.connect(host, port, timeout, retries - 1)
```

Comment:
Don't you think using recursion to mimic a loop increases the [WTFs/minute](https://i2.wp.com/commadot.com/wp-content/uploads/2009/02/wtf.png?w=550)? 😇

Reply:
Why? A lot less code and the semantics are cleaner?

Reply:
Hiding the concept of a loop behind recursion doesn't make it clearer to me, and using 2 lines of code instead of ~5 is also not "a lot less code" but ~3 lines less code :P

Reply:
You probably don't like haskell then :D